src/runtime/distributed/coordinator/kernels/Broadcast.h

```diff
@@ -0,0 +1,155 @@
+
+#include <runtime/local/datastructures/AllocationDescriptorGRPC.h>
+#include <runtime/local/datastructures/DataPlacement.h>
+#include <runtime/local/datastructures/Range.h>
+
+#include <cassert>
+#include <cstddef>
+
+// ****************************************************************************
+// Struct for partial template specialization
+// ****************************************************************************
+
+template<ALLOCATION_TYPE AT, class DT>
+struct Broadcast {
+    static void apply(DT *mat, bool isScalar, DCTX(ctx)) = delete;
+};
+
+// ****************************************************************************
+// Convenience function
+// ****************************************************************************
+
+template<ALLOCATION_TYPE AT, class DT>
+void broadcast(DT *&mat, bool isScalar, DCTX(ctx))
+{
+    Broadcast<AT, DT>::apply(mat, isScalar, ctx);
+}
+
+
+// ****************************************************************************
+// (Partial) template specializations for different distributed backends
+// ****************************************************************************
+
+// ----------------------------------------------------------------------------
+// GRPC
+// ----------------------------------------------------------------------------
+
+template<class DT>
+struct Broadcast<ALLOCATION_TYPE::DIST_GRPC, DT>
+{
+    static void apply(DT *&mat, bool isScalar, DCTX(ctx)) 
+    {
+        struct StoredInfo {
+            size_t dp_id;
+        };
+        DistributedGRPCCaller<StoredInfo, distributed::Data, distributed::StoredData> caller;
+        
+        
+        auto envVar = std::getenv("DISTRIBUTED_WORKERS");
+        assert(envVar && "Environment variable has to be set");
+        std::string workersStr(envVar);
+        std::string delimiter(",");
+
+        size_t pos;
+        std::vector<std::string> workers;
+        while ((pos = workersStr.find(delimiter)) != std::string::npos) {
+            workers.push_back(workersStr.substr(0, pos));
+            workersStr.erase(0, pos + delimiter.size());
+        }
+        workers.push_back(workersStr);
+
+        distributed::Data protoMsg;
+
+        assert(mat != nullptr && "Matrix to broadcast is nullptr");
+        double *val;
+        if (isScalar) {
+            auto ptr = (double*)(&mat);
+            val = ptr;
+            // Need matrix for metadata
+            mat = DataObjectFactory::create<DenseMatrix<double>>(0, 0, false); 
+            auto protoVal = protoMsg.mutable_value();
+            protoVal->set_f64(*val);
+        } 
+        else { // Not scalar
+            auto denseMat = dynamic_cast<const DenseMatrix<double>*>(mat);
+            if (!denseMat){
+                throw std::runtime_error("Distribute grpc only supports DenseMatrix<double> for now");
+            }
+            ProtoDataConverter<DenseMatrix<double>>::convertToProto(denseMat, protoMsg.mutable_matrix());
```

Comment:
The issue here is that `DistributedWrapper.h` that makes the call to `Distribute/Broadcast` use a `Structure` object, rather than `Dense/CSRMatrix`, which the `ProtoDataConverter` can handle/convert. Nonetheless, soon we will have our own implementation of serializing DAPHNE objects and we won't depend on `ProtoDataConverter` anymore, since we will simply transfer `bytes` using gRPC, instead of encoding matrices using `protobuf` representations, therefore this will be removed.

Reply:
OK then let's leave this as it is for now